### PR TITLE
feat: add inquiry and inquiry request page

### PR DESCRIPTION
**Pull request by vkdnjdldnjsw**

### CODE REVIEW

File: src/api/inquiry.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  email: string
+  title: string
+  body: string
+  answer?: string
+  uploaded_at: string
+  answered_at?: string
+}
+
+export class InquiryApi {
+  static async inquirys(
```
Comment (jhaemin) [inline src/api/inquiry.ts]: typo

File: src/pages/inquiry/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  function isReachBottom(): boolean {
+    return window.innerHeight < document.body.scrollHeight
+  }
+
+  async function loadMore(): Promise<void> {
+    if (isFetchRef.current) {
+      return
+    }
+    setIsFetching(true)
+    const moreItems = await InquiryApi.inquirys(itemsRef.current.length)
```
Comment (jhaemin) [inline src/pages/inquiry/index.tsx]: typo

File: src/pages/inquiry/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                              formatDate(new Date(item.answered_at)) +
+                              ')'
+                            : '답변대기 중'}
+                        </p>
+                      </div>
+                    </ArrowBlock>
+                  )
+                })
+              ) : (
+                <p className="no-inquirys">문의 내역이 없습니다.</p>
```
Comment (jhaemin) [inline src/pages/inquiry/index.tsx]: typo

File: src/pages/inquiry/index.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+              )}
+            </Grid>
+          </div>
+        </BaseLayout>
+      )}
+    </>
+  )
+}
+InquiryPage.getInitialProps = async (): Promise<InquiryProps> => {
+  const data = await InquiryApi.inquirys(0)
```
Comment (jhaemin) [inline src/pages/inquiry/index.tsx]: typo

File: src/pages/inquiry/request/index.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      alert('Error')
+    }
+  }
+  useEffect(() => {
+    focusTitle()
+  }, [])
+
+  return (
+    <>
+      <Head>문의하기</Head>
```
Comment (jhaemin) [inline src/pages/inquiry/request/index.tsx]: Wrong `Head` usage.
It should be
```tsx
<Head>
  <title>문의하기</title>
</Head>
```

File: src/pages/inquiry/request/index.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            placeholder="문의내용"
+            value={body}
+            onChange={(e): void => {
+              setBody(e.target.value)
+            }}
+            onKeyDown={(e): void => {
+              if (e.key === 'Enter') {
+                submit()
+              }
+            }}
```
Comment (jhaemin) [inline src/pages/inquiry/request/index.tsx]: It prevents line breaks inside `textarea`.

File: src/pages/inquiry/index.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    '-' +
+    ('0' + date.getDate()).slice(-2) +
+    ' ' +
+    ('0' + date.getHours()).slice(-2) +
+    ':' +
+    ('0' + date.getMinutes()).slice(-2) +
+    ':' +
+    ('0' + date.getSeconds()).slice(-2)
+  )
+}
```
Comment (jhaemin) [inline src/pages/inquiry/index.tsx]: Use `dayjs` module instead.

File: src/api/inquiry.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: `title=${title}&body=${body}&email=${email}`,
+    })
+    if (status === 201) {
+      return true
+    }
+    return false
+  }
+}
```
Comment (jhaemin) [inline src/api/inquiry.ts:70]: `useFetch` is deprecated in favor of `eodiroAxios`. For more information, checkout the source `eodiro-axios.ts` and `AuthApi` which is using the module.

File: src/pages/inquiry/request/index.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    }
+    setIsSbumitted(true)
+    const result = await InquiryApi.post(title, body, email)
+    if (result) {
+      window.location.href = '/inquiry'
+      alert('Submitted')
+    } else {
+      setIsSbumitted(false)
+      alert('Error')
+    }
```
Comment (jhaemin) [inline src/pages/inquiry/request/index.tsx]: You need to handle errors more specifically.
An email format error, for instance.

File: src/pages/inquiry/index.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    const moreItems = await InquiryApi.inquirys(itemsRef.current.length)
+    if (moreItems.inquiries && moreItems.inquiries.length > 0) {
+      setItems([...itemsRef.current, ...moreItems.inquiries])
+    }
+    setIsFetching(false)
+  }
+
+  useEffect(() => {
+    if (inquiries === undefined) {
+      Router.push('/inquiry/request')
```
Comment (jhaemin) [inline src/pages/inquiry/index.tsx]: Redirections should be processed on the server side for better SEO.
Use `getInitialProps`.

File: src/api/inquiry.ts (hunk 1 above)
Comment (vkdnjdldnjsw) [reply src/api/inquiry.ts]: done

File: src/pages/inquiry/index.tsx (hunk 2 above)
Comment (vkdnjdldnjsw) [reply src/pages/inquiry/index.tsx]: done

File: src/pages/inquiry/index.tsx (hunk 3 above)
Comment (vkdnjdldnjsw) [reply src/pages/inquiry/index.tsx]: done

File: src/pages/inquiry/index.tsx (hunk 4 above)
Comment (vkdnjdldnjsw) [reply src/pages/inquiry/index.tsx]: done

File: src/pages/inquiry/request/index.tsx (hunk 5 above)
Comment (vkdnjdldnjsw) [reply src/pages/inquiry/request/index.tsx]: done

File: src/pages/inquiry/request/index.tsx (hunk 6 above)
Comment (vkdnjdldnjsw) [reply src/pages/inquiry/request/index.tsx]: done

File: src/pages/inquiry/index.tsx (hunk 7 above)
Comment (vkdnjdldnjsw) [reply src/pages/inquiry/index.tsx]: done

**Review by jhaemin — APPROVED** [review]

**Pull request merged**